tests/integration/relations/test_relations.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+import asyncio
+import logging
+
+import psycopg2
+import pytest
+from pytest_operator.plugin import OpsTest
+from tenacity import Retrying, stop_after_delay, wait_fixed
+
+from ..helpers import CHARM_SERIES, METADATA
+from ..new_relations.test_new_relations import APPLICATION_APP_NAME, build_connection_string
+from ..relations.helpers import get_legacy_db_connection_str
+
+logger = logging.getLogger(__name__)
+
+APP_NAME = METADATA["name"]
+MAILMAN3_CORE_APP_NAME = "mailman3-core"
+DB_RELATION = "db"
+DATABASE_RELATION = "database"
+FIRST_DATABASE_RELATION = "first-database"
+APP_NAMES = [APP_NAME, APPLICATION_APP_NAME, MAILMAN3_CORE_APP_NAME]
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_deploy_charms(ops_test: OpsTest, charm):
+    """Deploy both charms (application and database) to use in the tests."""
+    # Deploy both charms (multiple units for each application to test that later they correctly
+    # set data in the relation application databag using only the leader unit).
+    async with ops_test.fast_forward():
+        await asyncio.gather(
+            ops_test.model.deploy(
+                APPLICATION_APP_NAME,
+                application_name=APPLICATION_APP_NAME,
+                num_units=1,
+                series=CHARM_SERIES,
+                channel="edge",
+            ),
+            ops_test.model.deploy(
+                charm,
+                application_name=APP_NAME,
+                num_units=1,
+                series=CHARM_SERIES,
+                config={"profile": "testing"},
+            ),
+            ops_test.model.deploy(
+                MAILMAN3_CORE_APP_NAME,
+                application_name=MAILMAN3_CORE_APP_NAME,
+                channel="stable",
```

Comment:
This charm has no updates for ages... and I would recommend to use 2 postgresql-test-app instead... but let's see how stable the current approach is.